packages/store-sdk/src/analytics/events.ts

```diff
@@ -0,0 +1,185 @@
+
+export interface ProductViewData {
+  product: any
+}
+
+export interface ProductClickData {
+  pageType: PageType
+  term?: string | undefined | null
+  position?: number | undefined | null
+  product: AnalyticsProduct
+}
+
+export interface ProductImpressionData {
+  impressions: Impression[]
+  product?: any // deprecated, use impressions list!
+  position?: number // deprecated, use impressions list!
+  list: string
+}
+
+export interface CartLoadedData {
+  orderForm: any
+}
+
+interface Order {
+  accountName: string
+  corporateName: string
+  coupon: string
+  currency: string
+  openTextField: string
+  orderGroup: string
+  salesChannel: string
+  visitorAddressCity: string
+  visitorAddressComplement: string
+  visitorAddressCountry: string
+  visitorAddressNeighborhood: string
+  visitorAddressNumber: string
+  visitorAddressPostalCode: string
+  visitorAddressState: string
+  visitorAddressStreet: string
+  visitorContactInfo: string[]
+  visitorContactPhone: string
+  visitorType: string
+  transactionId: string
+  transactionDate: string
+  transactionAffiliation: string
+  transactionTotal: number
+  transactionShipping: number
+  transactionSubtotal: number
+  transactionDiscounts: number
+  transactionTax: number
+  transactionCurrency: string
+  transactionPaymentType: PaymentType[]
+  transactionShippingMethod: ShippingMethod[]
+  transactionLatestShippingEstimate: Date
+  transactionProducts: ProductOrder[]
+  transactionPayment: {
+    id: string
+  }
+}
+
+interface Impression {
+  product: any
+  position: number
+}
+
+interface PaymentType {
+  group: string
+  paymentSystemName: string
+  installments: number
+  value: number
+}
+
+interface ShippingMethod {
+  itemId: string
+  selectedSla: string
+}
+
+export interface ProductOrder {
+  id: string
+  name: string
+  sku: string
+  skuRefId: string
+  skuName: string
+  productRefId: string
+  ean: string
+  slug: string
+  brand: string
+  brandId: string
+  seller: string
+  sellerId: string
+  category: string
+  categoryId: string
+  categoryTree: string[]
+  categoryIdTree: string[]
+  priceTags: PriceTag[]
+  originalPrice: number
+  price: number
+  sellingPrice: number
+  tax: number
+  quantity: number
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  components: any[]
+  measurementUnit: string
+  unitMultiplier: number
+}
+
+interface PriceTag {
+  identifier: string
+  isPercentual: boolean
+  value: number
+}
+
+export interface AnalyticsProduct {
+  productId: string
+  productReferenceId: string | null | undefined
+  productName: string
+  brand: string
+  categoryTree: Array<{ name: string }>
+  price: number
+  // TODO currencyCode
+  skuId: string
+  skuName: string
+  skuReferenceId: Array<{ value: string | null | undefined }> | null | undefined
+}
+
+export interface CartPixelProduct extends AnalyticsProduct {
+  quantity: number
+}
+
+export type PageType = 'fullTextSearch' | 'nonFullTextSearch' | 'other'
```

Comment:
what's the difference between nonFullText and other?

Reply:
It's a PLP but not full text. Such as category pages, brand pages, etc. Other can be home page, product page, or any other custom page.